traits/src/key_store.rs

```diff
@@ -13,9 +13,36 @@ pub trait ToKeyStoreValue {
 }
 
 /// The Key Store trait
-pub trait OpenMlsKeyStore: Send + Sync {
+pub trait OpenMlsKeyStore: Send + Sync + Debug {
     /// The error type returned by the [`OpenMlsKeyStore`].
-    type Error: std::error::Error + Debug;
+    type Error: std::error::Error + Debug + PartialEq + Clone;
```

Comment:
Since a lot of the errors in the MlsGroup API depend on `KeyStore::Error` now, we need these bounds (or at least `PartialEq`) to check in tests that we get the right errors. Not sure if there is a way around that, but I'm open to suggestions. (I did remove `Clone`, though.)

Reply:
The problem here is that this makes troube if the error doesn't implement these traits.
Generally the trait bounds should be kept as minimal as possible. The requirement is when calling the functions later and using the error. So ideally we'd be adding the requirements there to the type. But I think this would only work if we'd had a trait bound for  it here.
I'd be fine with adding `PartialEq` here again for now. But I expect that we should remove it again at some point.

Reply:
I understand. I've removed `Clone`, but left in `PartialEq` for now.

Reply:
The biggest issue with `PartialEq` on errors is that any error that contains (deep in the error chain or not) a `Backtrace` cannot be used, as `std::backtrace::Backtrace` doesn't implement `PartialEq` (https://doc.rust-lang.org/std/backtrace/struct.Backtrace.html#trait-implementations) and never will. As stacktraces are always unique, you cannot compare them.

If you want to assert that you get a certain error in tests you could always use `assert!(matches!(err, MyError::MyVariant(_)))` or something along those lines.

Reply:
Fair point. Using `matches` is a good idea for tests. I won't comb through all the tests in this PR, though, as it's already big enough. Do you want to create an issue for that? Should be easy enough to do.